agent_gateway/gateway/gateway.py

```diff
@@ -259,6 +259,13 @@ def _extract_answer(self, raw_answer):
             else:
                 return None
 
+    def _extract_replan_message(self, raw_answer):
+        replan_index = raw_answer.find("Action: Replan(") + len("Action: Replan(")
+        if replan_index != -1:
+            return raw_answer[replan_index : raw_answer.rfind(")")].strip()
+        else:
+            return "We couldn't find the information you're looking for. You can try rephrasing your request or validate that the provided tools contain sufficient information."
+
```

Comment:
```suggestion
    def _extract_replan_message(self, raw_answer):
        replan_start = "Action: Replan("
        replan_index = raw_answer.find(replan_start)
        if replan_index != -1:
            replan_index += len(replan_start)
            return raw_answer[replan_index : raw_answer.rfind(")")].strip()
        return (
            "We couldn't find the information you're looking for. You can try "
            "rephrasing your request or validate that the provided tools contain "
            "sufficient information."
        )
```

- Defines the text we're searching for as `replan_start`
- Check if `replan_start` is found in raw_answer before adding its length to replan_index.
- Moves the length check to be performed only if the string can be found
- Formats the return string to reduce column width